### DISCUSSION TITLE
Refresh all child data providers

### DISCUSSION THREAD

**Discussion by Valdri** [Budibase/budibase#6484] [Help] [answered]
This is a scheme of my application to display orders and then each item in that order.
-Data Provider1 - mongodb query for downloading orders
-button refresh data provider = Data Provider1
--Repeater
---Data Provider 2 mongodb query for downloading order-items

When I press the refresh button, only data provider1 is refreshed, data provider2 is not refreshed.

**Comment by melohagan** [comment, marked as answer]
Hi @Valdri 

Only the data provider you specify to be refreshed will be so.

It is worth noting that if you want the mongodb query to re-execute, then you can use a binding. When the value of a binding that is attached to a data provider filter, or that is being passed into a query, is changed, then it will automatically cause a refresh/re-execute that data provider/query.

**Reply by tmdoit** [reply]
Hi @melohagan,
Is it possible to turn off automatic data provider refresh when bind field is changed? Right now If I need to pass three parameters to the query, database receives two useless queries (each time when I change active input field).

**Reply by melohagan** [reply]
This answer may be useful to you! https://github.com/Budibase/budibase/discussions/10402

**Reply by tmdoit** [reply]
It helped, thank you! :)

**Reply by melohagan** [reply]
You're welcome 😄
